pkg/controllerutil/image_util.go

```diff
@@ -0,0 +1,202 @@
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package controllerutil
+
+import (
+	//  Import these crypto algorithm so that the image parser can work with digest
+	_ "crypto/sha256"
+	_ "crypto/sha512"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/distribution/reference"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/apecloud/kubeblocks/pkg/constant"
+	viper "github.com/apecloud/kubeblocks/pkg/viperx"
+)
+
+var imageLogger = log.Log.WithName("ImageUtil")
+
+type RegistryConfig struct {
+	From                     string `mapstructure:"from"`
+	To                       string `mapstructure:"to"`
+	RegistryDefaultNamespace string `mapstructure:"registryDefaultNamespace"`
+
+	// Quote from https://docs.docker.com/reference/cli/docker/image/tag/
+	// > While the OCI Distribution Specification supports more than two slash-separated components,
+	// > most registries only support two slash-separated components.
+	// > For Docker's public registry, the path format is as follows:
+	// > [NAMESPACE/]REPOSITORY: The first, optional component is typically a user's or an organization's
+	// > namespace. The second, mandatory component is the repository name. When the namespace is
+	// > not present, Docker uses `library` as the default namespace.
+	//
+	// So if there are more than two components (like `a/b` as a namespace), specify them both,
+	// or they won't be matched. Note empty namespace is legal too.
+	//
+	// key is the orignal namespace, value is the new namespace
+	NamespaceMapping map[string]string `mapstructure:"namespaceMapping"`
+}
+
+type RegistriesConfig struct {
+	DefaultRegistry  string           `mapstructure:"defaultRegistry"`
+	DefaultNamespace string           `mapstructure:"defaultNamespace"`
+	RegistryConfig   []RegistryConfig `mapstructure:"registryConfig"`
+}
+
+// this lock protects r/w to this variable itself,
+// not the data it points to
+var registriesConfigMutex sync.RWMutex
+var registriesConfig = &RegistriesConfig{}
+
+func GetRegistriesConfig() *RegistriesConfig {
+	registriesConfigMutex.RLock()
+	defer registriesConfigMutex.RUnlock()
+
+	// this will return a copy of the pointer
+	return registriesConfig
+}
+
+func ReloadRegistryConfig() {
+	registriesConfigMutex.Lock()
+	defer registriesConfigMutex.Unlock()
+
+	registriesConfig = &RegistriesConfig{}
+	if err := viper.UnmarshalKey(constant.CfgRegistries, &registriesConfig); err != nil {
```

Comment:
Does it return an error if this key is not found?

Reply:
No. In that case, `registriesConfig` will remain unchanged (which is an empty struct)